app/src/androidTest/java/org/mozilla/fenix/helpers/HomeActivityTestRule.kt

```diff
@@ -27,32 +27,59 @@ class HomeActivityTestRule(
     initialTouchMode: Boolean = false,
     launchActivity: Boolean = true,
     private val skipOnboarding: Boolean = false,
-) :
-    ActivityTestRule<HomeActivity>(HomeActivity::class.java, initialTouchMode, launchActivity) {
-
-    /**
-     * Helper for updating various app settings that could interfere with the tests.
-     * Tests that use [HomeActivityTestRule] are expected to rely on this [FeatureSettingsHelper]
-     * instead of instantiating their own.
-     *
-     * The main benefit this brings is better ordering of operations with the settings cleanup
-     * automatically happening just before the [Activity] under test finishes which may as opposed to
-     * cleanup happening earlier and modifying the app behavior.
-     */
-    val featureSettingsHelper = FeatureSettingsHelper()
+) : ActivityTestRule<HomeActivity>(HomeActivity::class.java, initialTouchMode, launchActivity),
+    FeatureSettingsHelper by FeatureSettingsHelperDelegate() {
+
+    // Using a secondary constructor allows us to easily delegate the settings to FeatureSettingsHelperDelegate.
+    // Otherwise if wanting to use the same names we would have to override these settings in the primary
+    // constructor and in that elide the FeatureSettingsHelperDelegate.
+    constructor(
+        initialTouchMode: Boolean = false,
+        launchActivity: Boolean = true,
+        skipOnboarding: Boolean = false,
+        isHomeOnboardingDialogEnabled: Boolean = settings.showHomeOnboardingDialog,
+        isPocketEnabled: Boolean = settings.showPocketRecommendationsFeature,
+        isJumpBackInCFREnabled: Boolean = settings.shouldShowJumpBackInCFR,
+        isRecentTabsFeatureEnabled: Boolean = settings.showRecentTabsFeature,
+        isRecentlyVisitedFeatureEnabled: Boolean = settings.historyMetadataUIFeature,
+        isPWAsPromptEnabled: Boolean = !settings.userKnowsAboutPwas,
+        isTCPCFREnabled: Boolean = settings.shouldShowTotalCookieProtectionCFR,
+        isWallpaperOnboardingEnabled: Boolean = settings.showWallpaperOnboarding,
+        isDeleteSitePermissionsEnabled: Boolean = settings.deleteSitePermissions,
+        etpPolicy: ETPPolicy = getETPPolicy(settings),
+    ) : this(initialTouchMode, launchActivity, skipOnboarding) {
+        this.isHomeOnboardingDialogEnabled = isHomeOnboardingDialogEnabled
+        this.isPocketEnabled = isPocketEnabled
+        this.isJumpBackInCFREnabled = isJumpBackInCFREnabled
+        this.isRecentTabsFeatureEnabled = isRecentTabsFeatureEnabled
+        this.isRecentlyVisitedFeatureEnabled = isRecentlyVisitedFeatureEnabled
+        this.isPWAsPromptEnabled = isPWAsPromptEnabled
+        this.isTCPCFREnabled = isTCPCFREnabled
+        this.isWallpaperOnboardingEnabled = isWallpaperOnboardingEnabled
+        this.isDeleteSitePermissionsEnabled = isDeleteSitePermissionsEnabled
+        this.etpPolicy = etpPolicy
+    }
```

Comment:
At least these 3 could be set as default to false so we don't have to change them in each test class. As far as I can remember we don't have tests for them yet.
 isJumpBackInCFREnabled = false,
 isTCPCFREnabled = false,
 isWallpaperOnboardingEnabled = false,

Reply:
An interesting problem arise if wanting to not have to explicitly change these flags.
In most of the cases this would lead to instantiating a rule without any parameter and in that use the default constructor which does not set the needed flags.
Don't see any way to get around this other than adding a new argument to the second constructor, an ugly throw-away one just to ensure the second constructor is used which will set some default flags values.
https://github.com/mozilla-mobile/fenix/blob/bd94db3fd3f70ebd5d51425823270731abcc3d55/app/src/androidTest/java/org/mozilla/fenix/helpers/HomeActivityTestRule.kt#L26-L62